src/aap_eda/api/serializers/activation.py

```diff
@@ -120,12 +115,18 @@ class Meta:
 
     def create(self, validated_data, user):
         self._validate_pre_reqs(user)
-        try:
-            rulebook = models.Rulebook.objects.get(
-                pk=validated_data["rulebook_id"]
-            )
-        except models.Rulebook.DoesNotExist:
-            raise IntegrityError
+        rulebook_id = validated_data["rulebook_id"]
+        decision_environment_id = validated_data["decision_environment_id"]
+        extra_var_id = validated_data.get("extra_var_id")
+
+        validate_activation_creation(
```

Comment:
This a response of [this comment](https://github.com/ansible/eda-server/pull/424#issuecomment-1745585351):
I'm not saying that we must perform field validation. This seems clearly an object validation case. I may missing something of your reasoning, my point is that such a validation should not be performed in the create method. We already make a call to the .is_valid() method as is the standard the way and part of this call is the object validation method "validate()". My point is that this logic should be there:
https://www.django-rest-framework.org/api-guide/serializers/#object-level-validation